ArchiSteamFarm/ArchiWebHandler.cs

```diff
@@ -446,6 +458,32 @@ internal sealed class ArchiWebHandler : IDisposable {
 			return await UrlGetToHtmlDocumentWithSession(SteamCommunityURL, request).ConfigureAwait(false);
 		}
 
+		internal async Task<HashSet<ulong>> GetDigitalGiftCards() {
+			const string requestGifts = SteamStoreURL + "/gifts";
+			HtmlDocument response = (await WebLimitRequest(SteamStoreURL, async () => await WebBrowser.UrlGetToHtmlDocument(requestGifts).ConfigureAwait(false)).ConfigureAwait(false))?.Content;
+			if (response == null) {
+				return null;
+			}
+
+			HtmlNodeCollection nodes = response.DocumentNode.SelectNodes("//div[@class='pending_gift']/div[starts-with(@id,'pending_gift_')]/div[@class='pending_giftcard_leftcol']/../@id");
```

Comment:
I don't see any reason for getting inner div just to return to outer one... Why not `//div[@class='pending_gift']/div[starts-with(@id,'pending_gift_')]/@id`?

Reply:
We need to check if `pending_giftcard_leftcol` is presented, because usual games gifts don't have this

Reply:
This makes sense then, isn't there ID somewhere inside? If not, it'd be a good idea to add appropriate comment stating that `pending_giftcard_leftcol` indicates gift card and why we're jumping to outer div from it. I'm also wondering if there is no xpath trick to just check if div has a child with that without moving to it, I'm not xpath expert myself either, so just a comment will do.